crates/kitsune_p2p/fetch/src/respond.rs

```diff
@@ -40,10 +40,14 @@ pub trait FetchResponseConfig: 'static + Send + Sync {
 }
 
 /// Manage responding to requests for data.
+#[derive(Clone)]
 pub struct FetchResponseQueue<C: FetchResponseConfig> {
     byte_limit: Arc<tokio::sync::Semaphore>,
     concurrent_send_limit: Arc<tokio::sync::Semaphore>,
     config: Arc<C>,
+    /// For testing, track the number of bytes sent.
+    #[cfg(feature = "test_utils")]
+    pub bytes_sent: Arc<std::sync::atomic::AtomicUsize>,
```

Comment:
Adding directly to the real type because this type is hard to substitute (https://github.com/rust-lang/rust/issues/20041) so I'd have to mess with quite a bit of code to support it. The problem with that is that test_utils is turned on by default... I think we should address that rather than this.